File: 1 basic/cpp/1004 .成绩排名 (20).cpp
#include<iostream>
#include<string>
using namespace std;

typedef struct{
	string name;
	string num;
	int score;
}Stud;


int main(){
	Stud st;
	string maxName,maxNum;
	string minName,minNum;
	int max=0,min=100,N;
	scanf("%d",&N);
	while(N--){
		cin>>st.name>>st.num>>st.score;
		if(max<st.score){
			max=st.score;
			maxName=st.name;
			maxNum=st.num;
		}
		if(min>st.score){
			min=st.score;
			minName=st.name;
			minNum=st.num;
		}
	}
	cout<<maxName<<' '<<maxNum<<endl;
	cout<<minName<<' '<<minNum<<endl;
	
	return 0;
}

File: 1 basic/py/1004. 成绩排名 (20).py
from macpath import split

n = int(input())
minName, minNum, minScore = '','',100
maxName, maxNum, maxScore = '','',0
for i in range(n):
    name,num, score = input().split()
    score = int(score)
    if score < minScore:
        minName = name
        minNum = num
        minScore = score
    if score > maxScore:
        maxName = name
        maxNum = num
        maxScore = score
print(maxName, maxNum)
print(minName, minNum)